cli/create/templates/templates.go

```diff
@@ -28,14 +33,76 @@ func init() {
 // Instantiate creates a file tree in a ctx.Project.Path according to ctx.Project.Template
 // It applies ctx.Project to the template
 func Instantiate(ctx *context.Ctx) error {
-	projectTmpl, exists := knownTemplates[ctx.Project.Template]
-	if !exists {
-		return fmt.Errorf("Template %s does not exists", ctx.Project.Template)
+	var err error
+	var projectTmpl *templates.FileTreeTemplate
+
+	if ctx.Create.From != "" {
+		log.Debugf("Template from %s is used", ctx.Create.From)
+
+		projectTmpl, err = parseTemplate(ctx.Create.From)
+		if err != nil {
+			return fmt.Errorf("Failed to parse template from specified path: %w", err)
+		}
+	} else {
+		var exists bool
+
+		log.Debugf("%s template is used", ctx.Create.Template)
+
+		projectTmpl, exists = knownTemplates[ctx.Create.Template]
+		if !exists {
+			return fmt.Errorf("Template %s does not exists", ctx.Create.Template)
+		}
 	}
 
 	if err := projectTmpl.Instantiate(ctx.Project.Path, ctx.Project); err != nil {
-		return fmt.Errorf("Failed to instantiate %s template: %s", ctx.Project.Template, err)
+		return fmt.Errorf("Failed to instantiate project template: %s", err)
 	}
 
 	return nil
 }
+
+func parseTemplate(from string) (*templates.FileTreeTemplate, error) {
+	if fileInfo, err := os.Stat(from); err != nil {
+		return nil, fmt.Errorf("Failed to use specified path: %s", err)
+	} else if !fileInfo.IsDir() {
+		return nil, fmt.Errorf("Specified path is not a directory: %s", from)
+	}
+
+	var tmpl templates.FileTreeTemplate
+
+	err := filepath.Walk(from, func(filePath string, fileInfo os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		relPath, err := filepath.Rel(from, filePath)
+		if err != nil {
+			return fmt.Errorf("Failed to get file path relative to the project root: %s", err)
+		}
+
+		if fileInfo.IsDir() {
+			tmpl.AddDirs(templates.DirTemplate{
+				Path: relPath,
+				Mode: fileInfo.Mode(),
+			})
+		} else {
+			fileContent, err := common.GetFileContent(filePath)
```

Comment:
Seems it could work quite slow for big projects. Especially if such project contains some C rocks with huge .so files